at_contacts_flutter/test/unit_tests/contact_screen_test.dart

```diff
@@ -21,5 +18,123 @@ void main() {
+          .thenAnswer((invocation) async => true);
+
+      ContactService().atContactImpl = mockAtcontactImpl;
+      var res = await ContactService()
+          .blockUnblockContact(contact: contact, blockAction: true);
+
+      expect(res, true);
+    });
+
+    test('mark_fav_contact', () async {
+      var contact = AtContact(atSign: "@83apedistinct");
+
+      when(() => mockAtcontactImpl.update(contact))
+          .thenAnswer((invocation) async => true);
+
+      ContactService().atContactImpl = mockAtcontactImpl;
+      var res = await ContactService().markFavContact(contact);
+
+      expect(res, true);
+    });
+
+    test('fetch_blocked_contact', () async {
+      var contact = AtContact(atSign: "@83apedistinct");
+
+      when(() => mockAtcontactImpl.listBlockedContacts())
+          .thenAnswer((invocation) async => [contact]);
+
+      ContactService().atContactImpl = mockAtcontactImpl;
+      var res = await ContactService().fetchBlockContactList();
+
+      expect(res, isA<List<AtContact>?>());
+    });
+
+    test("compare_blocked_contact_list_for_updates", () {
+      ContactService().blockContactList = [
+        AtContact(atSign: "@83apedistinct"),
+        AtContact(atSign: "@45expected")
+      ];
+      ContactService().baseBlockedList = [
+        BaseContact(AtContact(atSign: "@83apedistinct")),
+      ];
+
+      ContactService().compareBlockedContactListForUpdatedState();
+
+      expect(ContactService().baseBlockedList.length, 2);
+    });
+
+    test("delete_atSign", () async {
+      var atSign = "@83apedistinct";
+
+      when(() => mockAtcontactImpl.delete(atSign))
+          .thenAnswer((invocation) async => true);
+
+      ContactService().atContactImpl = mockAtcontactImpl;
+      var res = await ContactService().deleteAtSign(atSign: atSign);
+
+      expect(res, true);
+    });
+
+    test("remove_atSign", () async {
+      var contact = AtContact(atSign: "@83apedistinct");
+      ContactService().selectedContacts = [contact];
+      ContactService().removeSelectedAtSign(contact);
+
+      expect(ContactService().selectedContacts.length, 0);
+    });
+
+    test("select_atSign", () async {
+      var contact = AtContact(atSign: "@83apedistinct");
+      var contact2 = AtContact(atSign: "@45expected");
+      ContactService().selectedContacts = [contact];
+      ContactService().selectAtSign(contact2);
+
+      expect(ContactService().selectedContacts.length, 2);
+    });
+
+    test("clear_atSign", () async {
+      var contact = AtContact(atSign: "@83apedistinct");
+      var contact2 = AtContact(atSign: "@45expected");
+      ContactService().selectedContacts = [contact, contact2];
+      ContactService().clearAtSigns();
+
+      expect(ContactService().selectedContacts.length, 0);
+    });
+
+    test("compare_atSign", () async {
+      var atSign = "@45expected";
+      var atSign2 = "@83apedistinct";
+
+      var res = ContactService().compareAtSign(atSign, atSign2);
+      expect(res, false);
```

Comment:
Can we write a test for `compare_atSign` that will be true?

Reply:
okay